ownCloud/Media Uploads/PhotoKit Extensions/PHAsset+Upload.swift

```diff
@@ -456,41 +466,56 @@ extension PHAsset {
                 fileName = "\(self.ocStyleUploadFileName(with: sourceURL.deletingPathExtension().lastPathComponent.imageFileNameSuffix)).\(sourceURL.pathExtension)"
 			}
 
-			// Synchronously import media file into the OCCore and schedule upload
-			let importSemaphore = DispatchSemaphore(value: 0)
-
-			uploadProgress = sourceURL.upload(with: core,
-											  at: rootItem,
-											  alternativeName: fileName,
-											  modificationDate: self.creationDate,
-											  importByCopy: copySource,
-											  cellularSwitchIdentifier: cellularSwitchIdentifier,
-											  placeholderHandler: { (item, error) in
-												if !copySource && error != nil {
-													// Delete the temporary asset file in case of critical error
-													removeSourceFile()
-												}
-												if error != nil {
-													Log.error(tagged: ["MEDIA_UPLOAD"], "Sync engine import failed for asset ID \(self.localIdentifier)")
-												} else {
-													Log.debug(tagged: ["MEDIA_UPLOAD"], "Finished uploading asset ID \(self.localIdentifier)")
-												}
-												importResult = (item, error)
-												importSemaphore.signal()
-												uploadCompleteHandler?()
-
-			}, completionHandler: { (_, _) in
+			if let fpSession = fpSession {
+				fpSession.importThroughFileProvider(url: sourceURL, as: fileName, to: rootItem) { error in
+					importResult = (nil, error)
+					// Nothing to track
+					uploadProgress = Progress()
```

Comment:
The progress object created here is not used or returned anywhere.

Reply:
@felix-schwarz true.. removed it, anyway this part of code cries for refactoring